src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -158,15 +158,17 @@ namespace client
         virtual SelectAuthSchemeOptionOutcome SelectAuthSchemeOption(const AwsSmithyClientAsyncRequestContext& ctx) const = 0;
         virtual SigningOutcome SignRequest(std::shared_ptr<HttpRequest> httpRequest, const AuthSchemeOption& targetAuthSchemeOption) const = 0;
         virtual bool AdjustClockSkew(HttpResponseOutcome& outcome, const AuthSchemeOption& authSchemeOption) const = 0;
+        void SetFeatureHeaders(Aws::Http::HeaderValueCollection&& headers);
```

Comment:
while I think it is okay to add such field and member variable to the old AwsClient,
I can't say the same about new Smithy client.
On smithy client, we tried to avoid any service-specific customization from polluting the main request handling logic. In reality, only few generated clients will use this feature.

I'd suggest to insert this request header in the common service generated request class, such as
https://github.com/aws/aws-sdk-cpp/blob/main/generated/src/aws-cpp-sdk-sqs/include/aws/sqs/SQSRequest.h#L33
but there might be other options.